evolver/base.py

```diff
@@ -119,21 +119,22 @@ def model_validate(cls, obj, *, strict=None, from_attributes=None, context=None)
                 return cls.model_validate_json(obj, strict=strict, context=context)
             else:
                 # Cool, it was a ConfigDescriptor so validate from that. Remember ``ConfigDescriptor.config`` is just a
-                # dict.
+                # dict or None.
+                config_dict = {} if descriptor.config is None else descriptor.config
                 return super().model_validate(
-                    descriptor.config, strict=strict, from_attributes=from_attributes, context=context
+                    config_dict, strict=strict, from_attributes=from_attributes, context=context
```

Comment:
It seems to me that we can let the classes validate the input, and a None here would be allowed as per the new config type hint. I think the only place we need to handle the none is in construction...